mitodl/bootcamp-ecommerce
# feat: add sponsors

#### What are the relevant tickets?
https://github.com/mitodl/hq/issues/2775

#### What's this PR do?
- Adds an independent Sponsors page
- Displays the sponsors alongside the instructors
- Did some refactoring to reuse maximum code
- Some enhancements in slick
- Enhancements in CMS

#### How should this be manually tested?
- Setup a bootcamp page
- Open a bootcamp page in CMS, You should not see the `Sponsors Section` just like the `Instructor Sections`
- You should be able to add sponsors as well
- Open a bootcamp in frontend, App. You should see sponsors listed as well. These sponsors should be in independent blocks just like Instructors.
- Now, Add more than 4 Instructors and Sponsors and make sure the scroll slick works as usual

## Major Use cases to test

**Display and CMS pages**

1. Bootcamp page without any instructors and sponsors
2. Bootcamp page with instructors and without sponsors
3. Bootcamp page without any instructors but with sponsors
4. Bootcamp page with both instructors and sponsors
5. Bootcamp page with both instructors and sponsors having more than 4 entries so that you can test slick, carousel behavior.
6. Bootcamp page with both Multi-section instructors and Multisection sponsors

**Data validity and consistency**
- Check that there are no issues when this PR is being installed on an existing running instance from master (Setup from master branch, Setup pages)
- No issues in `seed_data` or `configure_cms` commands

#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
(Optional)

#### Screenshots (if appropriate)

**Sponsor Section in CMS (Bootcamp Run Page)**

<img width="1785" alt="add sponsors section (CMS)" src="https://github.com/mitodl/bootcamp-ecommerce/assets/34372316/c3f153d5-819e-411f-b107-eb6048d3f501">

**Added Sponsor Section in CMS (Bootcamp Run Page)**
<img width="1592" alt="image" src="https://github.com/mitodl/bootcamp-ecommerce/assets/34372316/d73ba3d2-3277-4c5c-a5e8-0c1abe8fc151">

---

Sponsors CMS Page Short look (Just like Instructors)

<img width="1589" alt="image" src="https://github.com/mitodl/bootcamp-ecommerce/assets/34372316/ef7c71f5-029e-41ec-98ba-fad913d6e354">

**Sponsors CMS Page Full look (Just like Instructors)**

![screencapture-bc-odl-local-8099-cms-pages-add-cms-sponsorssection-13-2023-10-26-19_15_04](https://github.com/mitodl/bootcamp-ecommerce/assets/34372316/2d2b5da3-6e27-4b16-a141-9a39dd7c835b)


---

**How it is displayed to the user**

<img width="455" alt="image" src="https://github.com/mitodl/bootcamp-ecommerce/assets/34372316/a0663b1f-875c-42f4-8408-c4d5f3e6f57f">


--- 

**How it is displayed to the user (Many items, Scroll, Instructors)**

<img width="553" alt="image" src="https://github.com/mitodl/bootcamp-ecommerce/assets/34372316/2c8cc718-cf7d-4ae2-92c6-59355ee0c1b7">

---
**How it is displayed to the user (Many items, Scroll, Sponsors)**
<img width="617" alt="image" src="https://github.com/mitodl/bootcamp-ecommerce/assets/34372316/cb9fa8f4-48f9-48df-af06-adf42adabebd">





#### What GIF best describes this PR or how it makes you feel?
(Optional)
